src/ansys/geometry/core/misc/checks.py

```diff
@@ -20,10 +20,61 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 """Provides functions for performing common checks."""
-from beartype.typing import Any, Optional, Tuple, Union
+from beartype.typing import TYPE_CHECKING, Any, Optional, Tuple, Union
 import numpy as np
 from pint import Unit
 
+if TYPE_CHECKING:
+    from ansys.geometry.core.designer import Design
+
+
+def ensure_design_is_active(method):
+    """
+    Make sure that the design is active before executing a method.
+
+    This function is necessary to be called whenever we do any operation on the design.
+    If we are just accessing information of the class, it is not necessary to call this.
+    """
+
+    def wrapper(self, *args, **kwargs):
+        # Check if the current design is active... otherwise activate it
```

Comment:
Minor comment suggestion
```suggestion
        # Check if the current design is active, otherwise activate it
```